### ISSUE TITLE
lights no longer working

### ISSUE THREAD

**Issue by ntalekt** [dwradcliffe/intellicenter#3]
Home Assistant 2024.3.3

Logger: homeassistant.components.light
Source: helpers/entity_platform.py:576
integration: Light ([documentation](https://www.home-assistant.io/integrations/light), [issues](https://github.com/home-assistant/core/issues?q=is%3Aissue+is%3Aopen+label%3A%22integration%3A+light%22))
First occurred: 7:15:48 PM (2 occurrences)
Last logged: 7:29:47 PM

Error adding entity light.lights for domain light with platform intellicenter

```
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 576, in _async_add_entities
    await coro
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 865, in _async_add_entity
    await entity.add_to_platform_finish()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1328, in add_to_platform_finish
    self.async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 992, in async_write_ha_state
    self._async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1113, in _async_write_ha_state
    state, attr, capabilities, shadowed_attr = self.__async_calculate_state()
                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1052, in __async_calculate_state
    attr.update(self.state_attributes or {})
                ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/light/__init__.py", line 1196, in state_attributes
    self.__validate_color_mode(color_mode, legacy_supported_color_modes, effect)
  File "/usr/src/homeassistant/homeassistant/components/light/__init__.py", line 1125, in __validate_color_mode
    effect_color_modes = supported_color_modes | {ColorMode.ONOFF}
                         ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~
TypeError: unsupported operand type(s) for |: 'ColorMode' and 'set'
```

light.lights (<class 'custom_components.intellicenter.light.PoolLight'>) does not report a color mode, this will stop working in Home Assistant Core 2025.3, please create a bug report at https://github.com/dwradcliffe/intellicenter/issues

**Comment by kracklow** [comment]
Same here unable to control pool light since upgrade.

**Comment by scottismyname** [comment]
Same, I see the warning in the log with "instructions" on what to do but it's not clear to me how to fix it exactly.

**Comment by scottismyname** [comment]
Figured out my issue, I had to make 2 changes, not sure if everyone else has to do the same.
1) When calling the service screenlogic.set_color_mode, I had to pass in config_entry instead of entry_id.
2) My entity id seems to have changed, not sure if it did for everyone.  Here is an example of my configuration using a custom button card:
![image](https://github.com/dwradcliffe/intellicenter/assets/9275733/09285ad7-209c-4bef-af80-49f0b60ddcfd)
